admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js

```diff
@@ -0,0 +1,177 @@
+
+const $ = window.$;
+
+export default class OrderViewPage {
+  constructor() {
+    this.orderProductManager = new OrderProductManager();
+    this.orderProductRenderer = new OrderProductRenderer();
+    this.orderPricesRefresher = new OrderPricesRefresher();
+  }
+
+  listenForProductDelete() {
+    $(OrderViewPageMap.productDeleteBtn).unbind('click').on('click', event => this.orderProductManager.handleDeleteProductEvent(event));
+
+    const callback = (event) => {
+      const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+      const numPages = $tablePagination.data('numPages');
+      const numRowsPerPage = $tablePagination.data('numPerPage');
+      const numRows = $(OrderViewPageMap.productsTable).find('tr[id^="orderProduct_"]').length;
+      let currentPage = parseInt($(OrderViewPageMap.productsTablePaginationActive).html(), 10);
+      const numProducts = parseInt($(OrderViewPageMap.productsCount).html(), 10);
+      if ((numProducts - 1) % numRowsPerPage === 0) {
+        this.orderProductRenderer.paginationRemovePage(numPages);
+      }
+      if (numRows === 1 && currentPage === numPages) {
+        currentPage -= 1;
+      }
+      this.orderProductManager.paginate(
+        event.orderId,
+        currentPage
+      );
+
+      this.orderProductRenderer.updateNumProducts(numProducts - 1);
+      this.orderPricesRefresher.refresh(event.orderId);
+    };
+
+    EventEmitter.off(OrderViewEventMap.productDeletedFromOrder, callback)
+      .on(OrderViewEventMap.productDeletedFromOrder, callback);
+  }
+
+  listenForProductEdit() {
+    $(OrderViewPageMap.productEditBtn).on('click', event => {
+      const $btn = $(event.currentTarget);
+      this.orderProductRenderer.moveProductsPanelToModificationPosition();
+      this.orderProductRenderer.editProductFromList(
+        $btn.data('orderDetailId'),
+        $btn.data('productQuantity'),
+        $btn.data('productPriceTaxIncl'),
+        $btn.data('productPriceTaxExcl'),
+        $btn.data('taxRate'),
+        $btn.data('location'),
+        $btn.data('availableQuantity'),
+      );
+    });
+    $(OrderViewPageMap.productCancelEditBtn).on('click', event => {
+      const $btn = $(event.currentTarget);
+      this.orderProductRenderer.resetEditRow($btn.data('orderDetailId'));
+      this.orderProductRenderer.moveProductPanelToOriginalPosition();
+    });
+
+    EventEmitter.on(OrderViewEventMap.productEditedToOrder, (event) => {
+      this.orderProductRenderer.addOrUpdateProductToList(event.orderDetailId, event.newRow);
+      this.orderProductRenderer.resetEditRow(event.orderDetailId);
+      this.orderPricesRefresher.refresh(event.orderId);
+      this.orderProductRenderer.moveProductPanelToOriginalPosition();
+      this.listenForProductDelete();
+      this.listenForProductEdit();
+    });
+  }
+
+  listenForProductAdd() {
+    $(OrderViewPageMap.productAddBtn).on('click', event => this.orderProductRenderer.moveProductsPanelToModificationPosition());
```

Comment:
```suggestion
    $(OrderViewPageMap.productAddBtn).on(
      'click', 
      event => this.orderProductRenderer.moveProductsPanelToModificationPosition()
    );
```
it's not readable, even on github :trollface: same below

Reply:
:heavy_check_mark: